domains/certificates/Query.API/API.IntegrationTests/Repositories/ContractStateTest.cs

```diff
@@ -0,0 +1,103 @@
+
+[Collection(IntegrationTestCollection.CollectionName)]
+public class ContractStateTest : IAsyncLifetime
+{
+    private ContractState _sut;
+    private readonly int _minimumAgeThresholdHours = 5;
+    private DbContextFactoryMock _dbContextFactoryMock;
+
+    public ContractStateTest(IntegrationTestFixture integrationTestFixture)
+    {
+        _dbContextFactoryMock = new DbContextFactoryMock();
+        var options = Options.Create(new MeasurementsSyncOptions() { MinimumAgeThresholdHours = _minimumAgeThresholdHours });
+        _sut = new ContractState(_dbContextFactoryMock, NullLogger<ContractState>.Instance, options);
+    }
+
+    [Theory]
+    [InlineData(-4, null, false)]
+    [InlineData(2, null, false)]
+    [InlineData(-4, -3, false)]
+    [InlineData(-6, -4, true)]
+    [InlineData(-7, -6, true)]
+    public async Task GivenContract_WhenGettingSyncInfo_ContractIsIncludedOrNot(int relativeContractStart, int? relativeContractEnd, bool isIncluded)
+    {
+        // Given contract
+        await using var dbContext = _dbContextFactoryMock.CreateDbContext();
+        var gsrn = Any.Gsrn();
+        var contractStart = UnixTimestamp.Now().AddHours(relativeContractStart);
+        var contractEnd = relativeContractEnd is null ? null : UnixTimestamp.Now().AddHours(relativeContractEnd.Value);
+        dbContext.Contracts.Add(Any.CertificateIssuingContract(gsrn, contractStart, contractEnd));
+        await dbContext.SaveChangesAsync();
+
+        // When getting sync info
+        var syncInfo = (await _sut.GetSyncInfos()).SingleOrDefault(si => si.Gsrn.Equals(gsrn));
+
+        // Contract is included or not
+        if (isIncluded)
+        {
+            syncInfo.Should().NotBeNull();
+            syncInfo!.StartSyncDate.Should().Be(contractStart.ToDateTimeOffset());
+        }
+        else
+        {
+            syncInfo.Should().BeNull();
+        }
+    }
+
+    [Fact]
+    public async Task GivenEndedContractWithMissingMeasurements_WhenGettingSyncInfo_ContractIsIncluded()
+    {
+        // Given ended contract with missing measurements
+        await using var dbContext = _dbContextFactoryMock.CreateDbContext();
+        var gsrn = Any.Gsrn();
+        var contractStart = UnixTimestamp.Now().AddHours(-20);
+        var contractEnd = UnixTimestamp.Now().AddHours(-10);
+        dbContext.Contracts.Add(Any.CertificateIssuingContract(gsrn, contractStart, contractEnd));
+        dbContext.MeteringPointTimeSeriesSlidingWindows.Add(MeteringPointTimeSeriesSlidingWindow.Create(gsrn, contractEnd,
+            [MeasurementInterval.Create(contractStart, contractStart.AddHours(1))]));
+        await dbContext.SaveChangesAsync();
+
+        // When getting sync info
+        var syncInfo = (await _sut.GetSyncInfos()).FirstOrDefault(si => si.Gsrn.Equals(gsrn));
+
+        // Contract is included
+        syncInfo.Should().NotBeNull();
+        syncInfo!.StartSyncDate.Should().Be(contractStart.ToDateTimeOffset());
```

Comment:
Er i tvivl. Returnerer contractstate egentligt null. eller empty list of syncinfos?